src/store/plugins/ui/currencies.js

```diff
@@ -0,0 +1,65 @@
+/* eslint no-param-reassign: ["error", { "ignorePropertyModificationsFor": ["state"] }] */
+
+import { timer } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import BigNumber from 'bignumber.js';
+import { fetchJson } from '../../utils';
+
+const currencies = {
+  usd: {
+    name: 'US Dollar',
+    symbol: '$',
+  },
+  eur: {
+    name: 'Euro',
+    symbol: '€',
+  },
+  btc: {
+    name: 'Bitcoin',
+    symbol: 'BTC',
+  },
+};
+
+const REFERENCE_CURRENCY = 'aeternity';
+const CURRENCIES = Object.keys(currencies).join(',');
+
+const rates$ = timer(0, 60000)
+  .pipe(
+    switchMap(() => fetchJson(
+      `https://api.coingecko.com/api/v3/simple/price?ids=${REFERENCE_CURRENCY}&vs_currencies=${CURRENCIES}`,
+    )),
+  );
+
+export default async (store) => {
+  const preferredCurrencyCode = 'eur';
+
+  store.registerModule('currencies', {
+    namespaced: true,
+    state: {
+      activeCode: (store.state.currencies && store.state.currencies.activeCode)
+        || (currencies[preferredCurrencyCode] ? preferredCurrencyCode : 'eur'),
+      rates: {},
+      swapped: false,
+    },
+    getters: {
+      list: () => Object.entries(currencies)
+        .map(([code, { name, symbol }]) => ({ code, name, symbol })),
+      active: ({ activeCode }, { list }) => list.find(({ code }) => code === activeCode),
+      rate: ({ activeCode, rates }) => (rates[REFERENCE_CURRENCY]
+        ? rates[REFERENCE_CURRENCY][activeCode] : BigNumber(0)),
+    },
+    mutations: {
+      setActiveCode(state, currencyCode) {
+        state.activeCode = currencyCode;
+      },
+      setRates(state, rates) {
+        state.rates = rates;
+      },
+      swapCurrencies(state) {
+        state.swapped = !state.swapped;
+      },
+    },
+  });
+
+  rates$.subscribe(r => store.commit('currencies/setRates', r));
```

Comment:
I propose to use this RxJs variable directly in components? and move it to observables.js